tests/unit/ui/components/TestRating.js

```diff
@@ -227,6 +227,51 @@ describe(__filename, () => {
     );
   });
 
+  it('selects stars on hover', () => {
+    const root = render();
+
+    const hoverStar = getStar({ root, rating: 4 });
+    hoverStar.simulate('mouseEnter', createFakeEvent());
+
+    // The first 4 should be selected:
+    for (const star of [1, 2, 3, 4]) {
+      expect(getStar({ root, rating: star })).toHaveClassName(
+        'Rating-selected-star',
+      );
+    }
+
+    // The last one should not be selected.
+    expect(getStar({ root, rating: 5 })).not.toHaveClassName(
+      'Rating-selected-star',
+    );
+  });
+
+  it('overrides the current rating when hovering over a star', () => {
+    const currentRating = 2;
+    const root = render({ rating: currentRating });
+
+    const hoverStar = getStar({ root, rating: 1 });
+    hoverStar.simulate('mouseEnter', createFakeEvent());
+
+    expect(getStar({ root, rating: currentRating })).not.toHaveClassName(
+      'Rating-selected-star',
+    );
+  });
+
+  it('finishes hovering on mouseLeave', () => {
+    const root = render();
+
+    const rating = 3;
+    const hoverStar = getStar({ root, rating });
+    hoverStar.simulate('mouseEnter', createFakeEvent());
+
```

Comment:
Would it be useful to have a check assertion in here that the star has the `Rating-selected-star` class, to avoid a possible false positive for the next assertion?

Reply:
I don't think it would be useful because that's a duplication of the previous test. The false positive wouldn't be an issue since the logic is already covered -- at least one test would fail.